sdk/core/Azure.Core/src/Shared/AzureSasCredentialSynchronousPolicy.cs

```diff
@@ -29,13 +29,26 @@ public override void OnSendingRequest(HttpMessage message)
             {
                 signature = signature.Substring(1);
             }
+
             if (!query.Contains(signature))
             {
-                query = string.IsNullOrEmpty(query) ? '?' + signature : query + '&' + signature;
+                // Get the signature history so we know which string to replace in the query if it was previously set.
+                if (message.TryGetProperty(typeof(AzureSasSignatureHistory), out object? previousSignature) && previousSignature != null & query.Contains((string)previousSignature!))
```

Comment:
nit: this is a lot of logical statements all together, would it be possible move into multiple statements for readability?

Reply:
They need to be one statement, but I'll break them into different lines.